File: frontend/frontend/src/modules/products/ProductService.js
import axios from 'axios';

const API_URL = 'http://localhost:8080/api/products';

export default {
  getAllProducts() {
    return axios.get(API_URL);
  },
  getProductById(productsid) {
    return axios.get(`${API_URL}/${productsid}`);
  },
  addProduct(product) {
    return axios.post(API_URL, product);
  },
  updateProduct(productsid, product) {
    return axios.put(`${API_URL}/${productsid}`, product);
  },
  deleteProduct(productsid) {
    return axios.delete(`${API_URL}/${productsid}`);
  },
};

File: demo/.mvn/src/main/java/com/example/demo/service/ProductService.java
package com.example.demo.service;

import java.util.List;

import com.example.demo.entity.Product;

public interface ProductService {

    List<Product> getAllProducts();

    Product getProductById(Integer productsid);

    void insertProduct(Product product);

    void updateProduct(Product product);

    void deleteProduct(Integer productsid);

}
